mm2src/mm2_p2p/src/behaviours/atomicdex.rs

```diff
@@ -824,11 +918,17 @@ fn maintain_connection_to_relays(swarm: &mut AtomicDexSwarm, bootstrap_addresses
                 if swarm.behaviour().core.gossipsub.is_connected_to_addr(&addr) {
                     continue;
                 }
```

Comment:
In practice, no, as it may cause a race condition on the network. This is how it was implement initially I believe and I would prefer to change this in the next libp2p refactoring (which will change pretty much whole connection and propagation layer) as it could introduce unknown side effects that we don't know about.